mail-attachment-downloader/src/main/java/com/baeldung/mail/attachment/utils/EmailAttachmentReceiver.java

```diff
@@ -0,0 +1,92 @@
+package com.baeldung.mail.attachment.utils;
+
+import static java.util.Arrays.asList;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+
+import javax.mail.Folder;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import javax.mail.NoSuchProviderException;
+import javax.mail.Part;
+import javax.mail.Session;
+import javax.mail.Store;
+import javax.mail.internet.MimeBodyPart;
+
+public class EmailAttachmentReceiver {
+
+    private static final String MAIL_POP3_PORT = "mail.pop3.port";
+    private static final String MAIL_POP3_HOST = "mail.pop3.host";
+    private static final String MAIL_POP3_SOCKET_FACTORY_PORT = "mail.pop3.socketFactory.port";
+    private static final String MAIL_POP3_SOCKET_FACTORY_FALLBACK = "mail.pop3.socketFactory.fallback";
+    private static final String MAIL_POP3_SOCKET_FACTORY_CLASS = "mail.pop3.socketFactory.class";
+    private static final String FALSE = "false";
+    private static final String JAVAX_NET_SSL_SSL_SOCKET_FACTORY = "javax.net.ssl.SSLSocketFactory";
+    private static final String POP_GMAIL_COM = "pop.gmail.com";
+    private static final String MULTIPART = "multipart";
+    private static final String INBOX = "INBOX";
+    private static final String POP3 = "pop3";
+    private static final int POP3_PORT = 995;
+
+    private String userName;
+    private String password;
+
+    public EmailAttachmentReceiver(String userName, String password) {
+        this.userName = userName;
+        this.password = password;
+    }
+
+    public String getUserName() {
+        return userName;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void downloadEmailAttachmentsTo(String directory) throws NoSuchProviderException, MessagingException, IOException {
```

Comment:
Let's not throw three exceptions from a method. That's generally bad Java practice.

Reply:
Done

Reply:
Done